### PR TITLE
highlightsToday true/false prop

### LINKED ISSUE #119: Setting textColor with mode prop set as date/datetime breaks to default text color
**the bug**
Setting `textColor` with `mode` prop set to `"date"` or `"datetime"` will incorrectly display "TODAY" as the default black color for light/white backgrounds, ignoring the `textColor`.

| props.mode   | props.textColor | Result                                                       |
| ------------ | --------------- | ------------------------------------------------------------ |
| not set      | `"#FFFFFF"`     | image |
| `"datetime"` | `"#FFFFFF"`     | image |
| `"date"`     | `"#FFFFFF"`     | image |
| `"time"`     | `"#FFFFFF"`     | image |

**Expected behavior**
Setting `textColor` with `mode` prop set to `"date"` or `"datetime"` will correctly set the current "Today" colour tint for the date picker component.

**To Reproduce**
Add example code that reproduces the behavior. 

```javascript
export default class App extends Component {
  state = { date: new Date() }
  render = () =>
    <DatePicker
      date={this.state.date}
      onDateChange={date => this.setState({ date })}
      mode="datetime"
      textColor="#FFFFFF"
    />
}
```

**Smartphone (please complete the following information):**
 - OS: iOS
 - React Native version 0.61.2
 - react-native-date-picker version 2.7.0

### PR DESCRIPTION
**Pull request by ptboyer**
Fixes #119 prevents date picker using today highlight color when setting mode to date/datetime with textColor also set.

~~As you can see it's appears to be fixed now!~~

UPDATE: Changing this pull request to add a prop instead, `highlightsToday` named in accordance with the iOS RNDatePicker property of the same name that controls whether or not to highlight Today's date in the picker with a different text color and changing the date string for that option to "Today".

With `mode="date"`, `textColor="#FFFFFF"` and `highlightsToday={false}`.

image

With `mode="datetime"`,`textColor="#FFFFFF"` and `highlightsToday={false}`.

image

### CODE REVIEW

**Comment by henninghall** [comment]
Hi thanks a lot for the pull request! With this solution, the "Today" string is removed right? That's not desirable behavior, see https://github.com/henninghall/react-native-date-picker/issues/72.

I think this issue might be related to dark mode instead, see https://github.com/henninghall/react-native-date-picker/issues/113

Could that be it?

**Comment by ptboyer** [comment]
Hey!

Upon closer inspection it appears that "Today" is indeed removed! I just noticed the text come back as the correct colour!

I think then this may be a case of the dark mode instead #113, however, looking at those solutions, it seems that a lot of then rely on iOS 13+ whereas I'm targeting from 9.2+.

I suspect that there is a property for setting the tint colour or highlight colour for the component.

And upon reading #72 it seems that my #119  issue is rediscovering the problem, however I will probe around now to see if I can find a solution.

Because ideally we want textColor to work for Today also, and maybe even find a way to set the Today to a custom tint colour for theming your own flavour of date picker.

**Comment by ptboyer** [comment]
@henninghall I've pushed a commit that adds a new prop `highlightsToday` that defaults to `true` and if set to `false` will `setHighlightsToday` to `NO` for the iOS date picker. That way it can be an opt in work around for people like me who aren't worried right now about the "Today" text string as a visual workaround for the today highlight breaking for potentially dark mode related stuff.

**Comment by ptboyer** [comment]
If you (@henninghall) are up for it, could you make the Android changes to make this prop cross platform? I added the prop description to the README also, but will leave screenshots to you with your style and formatting. 🙂

File: DatePickerIOS.js [hunk 1]
```diff
@@ -26,9 +26,10 @@ const RCTDatePickerIOS = requireNativeComponent('RNDatePicker')
  * source of truth.
  */
 export default class DatePickerIOS extends React.Component {
-  static DefaultProps = {
+  static defaultProps = {
```
Comment (ptboyer) [inline DatePickerIOS.js]: @henninghall Might be worth noting that this change is to fix a long over-looked bug that is even in the official react-native project, `DefaultProps` was never used as the correct `defaultProps`, which is why not setting `mode` actually was `undefined` and setting it to `"datetime"` produced weird side-effects as noticed in my issue.

**Comment by henninghall** [comment]
No need for highlightsToday prop after https://github.com/henninghall/react-native-date-picker/pull/148 is merged. It will automatically remove the today string when necessary (for other text colors than black or white)

**Pull request closed without merging**